src/main/java/org/ieeervce/gatekeeper/config/UserInfoUserDetails.java

```diff
@@ -2,31 +2,34 @@
 
 import org.ieeervce.gatekeeper.entity.User;
 import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Component;
 
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * implementation of UserDetails interface by spring security and mapping the data members to the user info
  */
 public class UserInfoUserDetails implements UserDetails {
-
-    private  String userEmail;
+    private  String userName;
     private  String password;
     private  boolean isEnabled;
     private List<GrantedAuthority> authorities;
 
     public UserInfoUserDetails(User userInfo){
-        userEmail = userInfo.getEmail();
```

Comment:
It may be simpler to store the `User` object, and just redirect it.
This way you don't store copies of the reference.

This way `UserInfoUserDetails` acts like an adapter for `User` to be used by Springboot

Ref: Java Composition